xdsl/interactive/app.py

```diff
@@ -102,6 +105,18 @@ def compose(self) -> ComposeResult:
 
         yield Footer()
 
+    @on(SelectionList.SelectedChanged)
+    def update_selected_view(self) -> None:
+        """
+        When the SelectionList (pass options) changes (i.e. a pass was selected or deselected), update the label to show
+        the query, and then call the update_current_module() function, which applies the selected passes to the input and displays the output
+        """
+        new_passes = "\n" + (", " + "\n").join(
+            p.name for p in self.passes_selection_list.selected
+        )
+        new_label = f"xdsl-opt -p {new_passes}"
+        self.query_one(Label).update(new_label)
```

Comment:
I am confused. Why is this now needed? Also, why are you saying that you are calling the `update_current_module`-function in the docu, but don't in the actual code?

Reply:
I agree here, that the documentation is confusing; and I think I've commented this on a previous PR.

If it is to use Textual's update model, I'd suggest avoiding documenting it as this is how the framework works and saves you from having to update 2 places in the source (functionality and documentation) if things change.